### DISCUSSION TITLE
How to set timeline in code?

### DISCUSSION THREAD

**Discussion by mechPenSketch** [dialogic-godot/dialogic#695] [Q&A]
Previous when dialog was a Singleton, I can call `Dialogic.set_current_timeline(timeline)` when I want to change timeline. Now, that method would not work. Neither does `Dialogic.start(timeline)`.

Maybe I supposed to call through the newly-defined dialog node:
```
dialogic_node = Dialogic.start(timeline)
add_child(dialogic_node)
```
However, I can't find the right method to change the timeline. Is there a way to change the timeline, or is it not implemented yet? Is it currently touched on? If not, I could work on it.

**Comment by Jowan-Spooner** [comment]
I'm not sure if I know what you want. 

### To simply start a timeline from code do
``` 
var dialog = Dialogic.start("MyTimelineName')
add_child(dialog)
```
You can also do 'MyFolder/MyTimeline' if that's how you folderstructure is set up.

### To start a timeline that was created through code do
```
var dialog = Dialogic.start('', '')
dialog.dialog_script = {
   "events":
      [
         { 'event_id':'dialogic_001', "character": "character-1621514281.json", "portrait":"", "text":"This was generated through code" }
      ]}
add_child(dialog)
```

**Reply by mechPenSketch** [reply]
I meant to change the timeline when the dialog is already added.

**Reply by Jowan-Spooner** [reply]
I see! In that case you will want to delete the previous dialog node and create a new one.

```
dialog.queue_free()
dialog = Dialogic.start('MyNextTimeline')
add_child(dialog)
```

**Reply by mechPenSketch** [reply]
With this approach, the dialog node will flash-disappear before starting the new timeline. I'm not satisfied with the result, but if this is the closest I could get, then perhaps I could make a new PR.

**Comment by mechPenSketch** [comment]
I've made a PR that lets me set the timeline through code: https://github.com/coppolaemilio/dialogic/pull/710
